components/mediation-admin/org.wso2.carbon.endpoint/src/main/java/org/wso2/carbon/endpoint/service/EndpointAdmin.java

```diff
@@ -198,6 +199,49 @@ public boolean addEndpoint(String epString) throws EndpointAdminException {
         return false;
     }
 
+    /**
+     * Set the tenant domain when a publisher publishes his API in MT mode. When publisher publishes
+     * the API, we login the gateway as supretenant. But we need to create the endpoint file in the
+     * particular tenant domain.
+     *
+     * @param endpointData Endpoint file content
+     * @param tenantDomain Domain of the logged tenant
+     * @return
+     * @throws EndpointAdminException Thrown if an error occur
+     */
+    public boolean addEndpointForTenant(String endpointData, String tenantDomain) throws EndpointAdminException {
+        try {
+            PrivilegedCarbonContext.startTenantFlow();
+            PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain,
+                    true);
+            boolean status = addEndpoint(endpointData);
+            return status;
```

Comment:
we can remove the status variable by calling "return addEndpoint(endpointData);"